### PR TITLE
fix: (managed-event-booking): slug conflict handling

### LINKED ISSUE #16363: [CAL-4173] Error when event type url already exists for user (managed event type)
When a new user is assigned to a managed event type but the user already has an event type with the same slug then it throws 'An unexpected Error occurred. Try again.' . Instead it should show the `ManagedEventDialog` 

Steps to reproduce: 
1. As a team admin create a new managed event type
2. Create a personal user event for the team admin with the same slug as the managed event type
3. In the assignment tab, assign a team admin to the managed event type
4. This should open the `ManagedEventDialog` but instead it throws an error

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-4173](https://linear.app/calcom/issue/CAL-4173/error-when-event-type-url-already-exists-for-user-managed-event-type)</sub>

### PR DESCRIPTION
**Pull request by abhijeetsingh-22**
## What does this PR do?

- Fixes #16363 
- Fixes CAL-4173

## ISSUE DEMO

https://github.com/user-attachments/assets/130f717d-1218-4cf2-8b23-4e5abc130666

## FIX DEMO
https://github.com/user-attachments/assets/ee8299b2-c580-4168-972e-381c36e405fd

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). N/A
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

Steps to test:

- As a team admin create a new managed event type
- Create a personal user event for the team admin or any other user with the same slug as the managed event type
- In the assignment tab, assign a user ( for which you created event in last step) to the managed event type
- This should open the `ManagedEventDialog`. Earlier it was throwing `An unexpected Error occurred. Try again`.

### CODE REVIEW

File: packages/lib/event-types/utils/getSlugConflictsInChildren.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  const conflicts = values.children.filter((child) => {
+    const isNewAssignment = !eventType.children.some((c) => c.owner.id === child.owner.id);
+    const isSlugPresentInChild = child.owner.eventTypeSlugs.includes(values.slug);
+    const isEventTypeSlugChanged = eventType.slug !== values.slug;
+
+    const isNewAssignmentWithConflict = isNewAssignment && isSlugPresentInChild;
+    const isExistingAssignmentWithConflict =
+      !isNewAssignment && isEventTypeSlugChanged && isSlugPresentInChild;
+    return isNewAssignmentWithConflict || isExistingAssignmentWithConflict;
+  });
```
Comment (Amit91848) [inline packages/lib/event-types/utils/getSlugConflictsInChildren.ts:19]: This works but why do we need to know if new users were assigned and conflicting or if parent event slug was changed and it is conflicting with existing users? Aren't we showing the same message.

Let's not complicate it and simply check if values.slug is present in children eventTypeSlugs.

Lmk if there is something I am missing here

**Review by Amit91848 — CHANGES_REQUESTED** [review]
Thanks for the PR @abhijeetsingh-22 , it works great! But I have some doubts with the code.

File: packages/lib/event-types/utils/getSlugConflictsInChildren.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  values,
+  eventType,
+}: {
+  values: FormValues;
+  eventType: EventTypeSetupProps["eventType"];
+}) => {
+  const conflicts = values.children.filter((child) => {
+    const isNewAssignment = !eventType.children.some((c) => c.owner.id === child.owner.id);
+    const isSlugPresentInChild = child.owner.eventTypeSlugs.includes(values.slug);
+    const isEventTypeSlugChanged = eventType.slug !== values.slug;
```
Comment (anikdhabal) [inline packages/lib/event-types/utils/getSlugConflictsInChildren.ts:13]: Why do we need to check if it’s a new assignment or not? Our main priority is to check if the slug is present in the child or not. You can simplify it by removing the extra steps

**Review by anikdhabal — DISMISSED** [review]

File: packages/lib/event-types/utils/getSlugConflictsInChildren.ts (hunk 1 above)
Comment (abhijeetsingh-22) [reply packages/lib/event-types/utils/getSlugConflictsInChildren.ts]: hey 
Thanks for the review. I have refactored the logic!!

File: packages/features/eventtypes/components/tabs/assignment/EventTeamAssignmentTab.tsx [hunk 3]
```diff
@@ -422,7 +422,12 @@ export const EventTeamAssignmentTab = ({
     .map((member) => mapUserToValue(member, t("pending")));
   const childrenEventTypeOptions = teamMembers.filter(pendingMembers).map((member) => {
     return mapMemberToChildrenOption(
-      { ...member, eventTypes: member.eventTypes.filter((et) => et !== eventType.slug) },
+      {
+        ...member,
+        eventTypes: member.eventTypes.filter(
+          (et) => et !== eventType.slug || !eventType.children.some((c) => c.owner.id === member.id)
+        ),
+      },
```
Comment (anikdhabal) [inline packages/features/eventtypes/components/tabs/assignment/EventTeamAssignmentTab.tsx:430]: I don't think this change needed

Comment (abhijeetsingh-22) [reply packages/features/eventtypes/components/tabs/assignment/EventTeamAssignmentTab.tsx:430]: It's required.
For user events where the event slug matches the `EventType` slug:
- If the user is already a member of the event, it ensures that the event is excluded from conflict checks.
- If the user is not a member of the event, the event is included in conflict checks.

**Review by Amit91848 — APPROVED** [review]
Tested and works like a charm ! LGTM 🔥🔥. Thanks for the PR @abhijeetsingh-22

**Pull request merged**